leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/send/SendTimestampedTest.java

```diff
@@ -0,0 +1,181 @@
+
+        for (Long ts : exampleNodes.keySet()) {
+            Map<LwM2mPath, LwM2mNode> pathNodeMap = data.getNodesForTimestamp(ts);
+            assertTrue(pathNodeMap.containsKey(getExamplePath()));
+
+            LwM2mNode node = pathNodeMap.get(getExamplePath());
+            LwM2mNode expectedNode = exampleNodes.get(ts);
+
+            assertEquals(node, expectedNode);
+        }
+    }
+
+    private static LwM2mPath getExamplePath() {
+        return new LwM2mPath("/2000/2/3");
+    }
+
+    private static Map<Long, LwM2mNode> getExampleTimestampedNodes() {
+        Map<Long, LwM2mNode> timestampedNodes = new HashMap<>();
+        timestampedNodes.put(2222L, LwM2mSingleResource.newIntegerResource(3, 12345));
+        timestampedNodes.put(4444L, LwM2mSingleResource.newIntegerResource(3, 67890));
+        return timestampedNodes;
+    }
+
+    private static class TestHelperWithFakeDecoder extends IntegrationTestHelper {
+        @Override
+        protected ObjectsInitializer createObjectsInitializer() {
+            return new ObjectsInitializer(new StaticModel(createObjectModels()));
+        }
+
+        @Override
+        public void createServer() {
+            LeshanServerBuilder serverBuilder = createServerBuilder();
+            serverBuilder.setDecoder(new FakeDecoder());
+            server = serverBuilder.build();
+            setupServerMonitoring();
+        }
+    };
+
+    private static class FakeDecoder implements LwM2mDecoder {
+        private final DefaultLwM2mDecoder decoder = new DefaultLwM2mDecoder(true);
+
+        @Override
+        public LwM2mNode decode(byte[] content, ContentFormat format, LwM2mPath path, LwM2mModel model)
+                    throws CodecException {
+            return decoder.decode(content, format, path, model);
+        }
+
+        @Override
+        public <T extends LwM2mNode> T decode(byte[] content, ContentFormat format, LwM2mPath path,
+                LwM2mModel model, Class<T> nodeClass) throws CodecException {
+            return decoder.decode(content, format, path, model, nodeClass);
+        }
+
+        @Override
+        public Map<LwM2mPath, LwM2mNode> decodeNodes(byte[] content, ContentFormat format, List<LwM2mPath> paths,
+                LwM2mModel model) throws CodecException {
+            return decoder.decodeNodes(content, format, paths, model);
+        }
+
+        @Override
+        public TimestampedLwM2mNodes decodeMultiTimestampedNodes(byte[] content, ContentFormat format,
+                LwM2mModel model) throws CodecException {
+            Map<Long, LwM2mNode> timestampedNodes = getExampleTimestampedNodes();
+
+            TimestampedLwM2mNodesImpl data = new TimestampedLwM2mNodesImpl();
+            for (Map.Entry<Long, LwM2mNode> entry : timestampedNodes.entrySet()) {
+                data.put(entry.getKey(), getExamplePath(), entry.getValue());
+            }
+
+            return data;
+        }
+
```

Comment:
I'm not sure I get the point here because it seems that the server will ignore the `byte[] content` ?
Should we rather hack the client instead of server ? 
I mean the purpose of this PR is to add the feature at server side without the client side.
So I can understand that we hack the client to make it send a timestampted payload but if we hack the server I'm not sure what we are testing here ? :thinking: